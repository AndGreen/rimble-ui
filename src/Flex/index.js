import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Box from '../Box';

const Flex = React.forwardRef((props, ref) => <Box ref={ref} {...props} />);

Flex.defaultProps = {
  display: 'flex',
};

Flex.propTypes = {
  ...Box.propTypes,
};

Flex.displayName = 'Flex';

export default Flex;
