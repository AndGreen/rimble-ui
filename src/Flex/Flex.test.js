import Flex from './'

describe('Flex component sanity', () => {
  it('has name', () => {
    expect(Flex.displayName).toBe('Flex')
  })

  it('matches default snapshot', () => {
    const component = render(<Flex />)
    expect(component).toMatchSnapshot()
  })
})
