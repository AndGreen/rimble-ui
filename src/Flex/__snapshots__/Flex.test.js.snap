// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  
</div>
`;
